File: src/main/java/com/inline/InLine/entity/QueueSession.java
package com.inline.InLine.entity;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class QueueSession {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String code;

    @Column(nullable = false)
    private Integer averageServiceMinutes;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private QueueStatus status;


    @PrePersist
    void beforeInsert (){
        if(status == null){
            status = QueueStatus.OPEN;
        }

        if(createdAt==null){
            createdAt = LocalDateTime.now();
        }
    }


}

File: mobile/lib/features/queue/model/queue_session.dart
enum QueueStatus { open, closed }

class QueueSession {
  const QueueSession({
    required this.id,
    required this.name,
    required this.code,
    required this.averageServiceMinutes,
    required this.createdAt,
    required this.status,
  });

  final int id;
  final String name;
  final String code;
  final int averageServiceMinutes;
  final QueueStatus status;
  final DateTime createdAt;

  factory QueueSession.fromJson(Map<String, dynamic> json) {
    return QueueSession(
      id: json["id"] as int,
      name: json["name"] as String,
      code: json["code"] as String,
      averageServiceMinutes: json["averageServiceMinutes"] as int,
      createdAt: DateTime.parse(json['createdAt'] as String),
      status: QueueStatus.values.byName(
        (json['status'] as String).toLowerCase(),
      ),
    );
  }
}
